### PR TITLE
Updated MassTransit instrumentation to OpenTelemetry 1.0

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
This PR upgrades the MassTransit instrumentation to OTel version 1.0.2

### CODE REVIEW

**Comment by alexvaluyskiy** [comment]
One test, related to trace filtering is failing (I will investigate why). But the rest of the PR is ready to review

**Review by cijothomas — APPROVED** [review]
Looks good from the perspective of removing ActivitySourceAdapter and leveraging AddLegacySource feature.

Not MT expert to offer deeper review :)

File: test/OpenTelemetry.Contrib.Instrumentation.MassTransit.Tests/MassTransitInstrumentationTests.cs [hunk 1]
```diff
@@ -261,9 +265,9 @@ public async Task MassTransitInstrumentationTestOptions()
                 }
             }
 
-            Assert.Equal(4, activityProcessor.Invocations.Count);
+            Assert.Equal(8, activityProcessor.Invocations.Count);
```
Comment (alexvaluyskiy) [inline test/OpenTelemetry.Contrib.Instrumentation.MassTransit.Tests/MassTransitInstrumentationTests.cs:268]: By default, the instrumentation should create 4 activities per message (publish, receive, consume, and handle). In this test, we are filtering tracing operations and should create only 1 activity. In the OpenTelemetry 0.7 it executed exactly one invocation. But in the current version, it returns 4 StartActivity events and 1 EndActivity event. 
@cijothomas  is it correct behavior?

Comment (utpilla) [reply test/OpenTelemetry.Contrib.Instrumentation.MassTransit.Tests/MassTransitInstrumentationTests.cs:268]: This is the correct behavior @alexvaluyskiy. With the new change to `TracerProviderSdk` to support legacy activities, the `Processor.OnStart` will be called for any legacy activity when it starts if it satisfies these conditions:
- `activity.OperationName` is added as a Legacy Source to `TracerProviderSdk`
- `activity.IsAllDataRequested` is `true` (this is set by the sampler)

When you set `activity.IsAllDataRequested` to `false` in the `OnStartActivity` method of the listener after filtering out the OperationNames that are not contained in `TracedOperations`, it ensures that `Processor.OnEnd` which is also responsible for exporting is not called for such activities.

You already have an `Assert` which ensures that there are exactly two invocations on the Processor with the required Activity OperationName.  If you'd like, you could go a little more specific in your testing to assert that the `Processor.OnEnd`  is only called for the activity whose OperationName is contained in `TracedOperations`. So in this test you could add something like this:
```
Assert.Single(activityProcessor.Invocations, invo => invo.Method.Name == "OnEnd");
var onEndProcessedActivity = activityProcessor.Invocations.FirstOrDefault(invo => invo.Method.Name == "OnEnd").Arguments[0] as Activity;
Assert.Equal(operationName, onEndProcessedActivity.OperationName);
```

Comment (alexvaluyskiy) [reply test/OpenTelemetry.Contrib.Instrumentation.MassTransit.Tests/MassTransitInstrumentationTests.cs:268]: Thank you for the explanation. I've changed the assertions

**Comment by alexvaluyskiy** [comment]
@cijothomas any blockers for this PR?

**Comment by cijothomas** [comment]
> @cijothomas any blockers for this PR?

Would you update codeowners file to list yourself as an approver for the MassTransit instrumentation? Similar to this https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CODEOWNERS#L12

File: src/OpenTelemetry.Contrib.Instrumentation.MassTransit/Implementation/MassTransitDiagnosticListener.cs [hunk 2]
```diff
@@ -18,48 +18,62 @@
...
-            this.activitySource.Start(activity, this.GetActivityKind(activity));
+            ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
+            ActivityInstrumentationHelper.SetKindProperty(activity, this.GetActivityKind(activity));
         }
 
         public override void OnStopActivity(Activity activity, object payload)
         {
             if (this.options.TracedOperations != null && !this.options.TracedOperations.Contains(activity.OperationName))
             {
+                MassTransitInstrumentationEventSource.Log.RequestIsFilteredOut(activity.OperationName);
```
Comment (cijothomas) [inline src/OpenTelemetry.Contrib.Instrumentation.MassTransit/Implementation/MassTransitDiagnosticListener.cs:61]: nit: `RequestIsFilteredOut` doesn't sound the apt name here.

**Comment by alexvaluyskiy** [comment]
@cijothomas thank you. I will fix remaining comments in the next pull requests.
What actions should be performed to release this instrumentation to nuget?

**Comment by cijothomas** [comment]
@alexvaluyskiy The release is manual, and one of the maintainers have to do it now.
@srprash Did you get a chance to document the process for releasing from this repo?

(@CodeBlanch did a manual release for wcf last week, but we need to list the process and (then automate))

**Comment by alexvaluyskiy** [comment]
But I think, we have to wait for OpenTelemetry 1.0.2 release first

**Comment by cijothomas** [comment]
> But I think, we have to wait for OpenTelemetry 1.0.2 release first

yea. we'll likely name it (OpenTelemetry .SDK)1.1.0-beta1 before pushing to nuget.org.

**Comment by CodeBlanch** [comment]
The release process I used for WCF...

# Setup

* Pick your favorite workflow from .github/workflows and copy/rename for your project. Update the workflow name, project name, and tag prefix. I wish we didn't have to copy/paste like this, but I looked around and github workflows don't support templates yet (from what I could tell).

* In your csproj add `<MinVerTagPrefix>Instrumentation.Wcf-</MinVerTagPrefix>` except with the tag prefix you picked. The tag prefix here needs to match what's in the workflow (minus wildcard).

# Release

* Push a tag to opentelemetry-dotnet-contrib using the tag prefix you selected and the version you want to release. It is a good idea to check with @cijothomas for the version to use because if you make a mistake you have to unlist the package and we can never use that version again.

   ```bash
   $ git tag -a Instrumentation.Wcf-1.0.0-rc1 -m Instrumentation.Wcf-1.0.0-rc1
   $ git push origin Instrumentation.Wcf-1.0.0-rc1
   ```

   I usually work on a fork of opentelemetry-dotnet-contrib but I ran those commands directly against a clone of the main repo. I'm sure there is a way to add tag on the fork origin and push to upstream, but I'm kind of git CLI newb so I went this route.

* The tag should kick off an automatic run of the workflow you configured. Check the "Actions" tab on GitHub. Wait for it to complete.

* The output of the workflow will be attached automatically to the run. Download it right from the GitHub UI.

* Upload the .nupkg and .snupkg from the downloaded zip to NuGet. For me NuGet automatically populated everything correctly. You need to have access on NuGet to do this.

**Comment by cijothomas** [comment]
thanks @CodeBlanch 
@srprash Please take a look at the steps Mikel followed above, when you writing releasing doc. This is missing the Github release, so we need to add that too.

**Pull request merged**